packages/amplify-provider-awscloudformation/src/cloud-formation-error-handler.ts

```diff
@@ -0,0 +1,34 @@
+import { $TSAny, AmplifyError } from 'amplify-cli-core';
+
+export const handleCloudFormationError = (err: Error & { details?: string }): void => {
+  if (err?.name === 'ValidationError' && err?.message === 'No updates are to be performed.') {
+    return;
+  }
+
+  if (err?.name === 'ValidationError' && (err?.message ?? '').includes('_IN_PROGRESS state and can not be updated.')) {
+    throw new AmplifyError(
+      'DeploymentInProgressError',
+      {
+        message: 'Deployment is already in progress.',
+        resolution: 'Wait for the other deployment to finish and try again.',
+        code: (err as $TSAny).code,
+      },
+      err,
+    );
+  }
+
+  const s3Indicator = '(AWS::S3::Bucket)';
+  const alreadyExistsSuffix = 'already exists';
+  if (err?.details?.includes(s3Indicator) && err?.details?.includes(alreadyExistsSuffix)) {
+    const prefix = `Name: ${
+      err.details.includes('CustomMessageConfirmationBucket') ? 'CustomMessageConfirmationBucket' : 'S3Bucket'
+    } (AWS::S3::Bucket), Event Type: create, Reason:`;
```

Comment:
Should this error handling be in the handleCloudFormation Error? This seems more specific to S3 errors. Would it be more maintainable to add a separate function that handles S3 errors and branch out from this function if those come up?

Reply:
Further, is there an error code associated with this error message? It seems like searching for that error string can lead to a breaking change in the future if S3 changes their error message?

Reply:
This is a CloudFormation error when deploying an S3 bucket. I think that it should go in this function since this is the function where all deployment errors are passed. I agree with you that if the error message is changed it could cause this error to go uncaught, but that is all the information in the error object that there is to work from.